### PR TITLE
[TGUI] Cargo Console

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Converts the cargo console to TGUI. It still functions exactly the same, but now looks a lot better, as well as having a ton of optimisations under the hood. The only notable changes are the fact you now have a searchbox, and theres now a log of messages CC sends to cargo, which was done for 2 reasons.

1: It is nice to have a historical record of cargo's performance throughout the shift
2: I was not about to snowflake the living hell out of a TGUI just to make it accept spans and other garbage

Sidenote; FOR SOME UNKNOWN REASON, THE PBULIC SUPPLY ORDERING CONSOLE IS **NOT** A SUBTYPE OF THE REGUALR ONE AND IT USED TO HAVE AHLF OF ITS CODE DUPLICATED. WHAT THE ACTUAL HELL.

Some code taken from the autolathe PR from @Kyep [#14225] 

## Why It's Good For The Game
TGUI Good. NanoUI Bad.

## Images of changes
Home screen, with 2 orders placed, and 2 pending
image

Searching for `Gun` (Search is case-insensitive)
image

Scrolling because of a mass amount of items and orders
image

Modal of crate contents
image

Public view
image
*Note lack of control buttons, and approval will always be greyed out. And yes, you can still only cancel your own orders*

CC message view
image

## Changelog
:cl: AffectedArc07
add: Cargo console is now TGUI. It even has a search box.
/:cl:

### CODE REVIEW

**Comment by TDSSS** [comment]
IMO, confirmed orders vs. requests could stand to be a bit easier to tell apart at a glance, but otherwise looks very good!

File: tgui/packages/tgui/interfaces/CargoConsole.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
+import { flow } from 'common/fp';
+import { Fragment } from 'inferno';
+import { filter, sortBy } from 'common/collections';
+import { useBackend, useSharedState, useLocalState } from "../backend";
+import { Button, LabeledList, Box, AnimatedNumber, Section, Dropdown, Input, Table, Modal } from "../components";
+import { Window } from "../layouts";
+import { LabeledListItem } from "../components/LabeledList";
+import { createSearch, toTitleCase } from 'common/string';
+
+export const CargoConsole = (props, context) => {
```
Comment (dearmochi) [inline tgui/packages/tgui/interfaces/CargoConsole.js:10]: Looks good but this needs to be broken into multiple components e.g `Status`, `Browse` and `Details`.

File: code/modules/shuttle/supply.dm [hunk 2]
```diff
@@ -547,45 +409,23 @@
...
-				// 0/1 after the pack name (set_name) is a boolean for ordering multiple crates
-				packs_list.Add(list(list("name" = pack.name, "amount" = pack.amount, "cost" = pack.cost, "command1" = list("doorder" = "[set_name]0"), "command2" = list("doorder" = "[set_name]1"), "command3" = list("contents" = set_name))))
-
-	data["supply_packs"] = packs_list
-	if(content_pack)
-		var/pack_name = sanitize(content_pack.name)
-		data["contents_name"] = pack_name
-		data["contents"] = content_pack.manifest
-		data["contents_access"] = content_pack.access ? get_access_desc(content_pack.access) : "None"
+/obj/machinery/computer/supplycomp/tgui_data(mob/user, ui_key = "main", datum/topic_state/state = GLOB.default_state)
```
Comment (dearmochi) [inline code/modules/shuttle/supply.dm:427]: `tgui_data(mob/user)`

**Review by dearmochi — APPROVED** [review]
LGTM, but I'm bothered by that empty space in the screenshots
image

**Comment by AffectedArc07** [comment]
> LGTM, but I'm bothered by that empty space in the screenshots
> image

Empty space is there because of table cells and things left aligning from the column. It offsets slightly since the text is a little wider than the buttons
image

File: tgui/packages/tgui/interfaces/CargoConsole.js (hunk 1 above)
Comment (AffectedArc07) [reply tgui/packages/tgui/interfaces/CargoConsole.js:10]: Any technical reason for this or just for styling

File: code/modules/shuttle/supply.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/modules/shuttle/supply.dm:427]: Will do

File: tgui/packages/tgui/interfaces/CargoConsole.js (hunk 1 above)
Comment (dearmochi) [reply tgui/packages/tgui/interfaces/CargoConsole.js:10]: Styling. Long functions aren't particularly agreeable to look at no matter the language.

**Comment by AffectedArc07** [comment]
image
Buttons now align

**Comment by dearmochi** [comment]
LGTM 👍

**Comment by Mitchs98** [comment]
This is beautiful.

**Review by variableundefined — APPROVED** [review]

**Pull request merged**